net/Socket.cpp

```diff
@@ -86,9 +102,57 @@ int Socket::createSocket([[maybe_unused]] Socket::Type type)
 #endif
 }
 
+const std::string Socket::getClientAddressAndPort() const noexcept
+{
+    std::string s;
+    if (Type::IPv6 == type())
+    {
+        s.append("[").append(clientAddress()).append("]:").append(std::to_string(clientPort()));
+    }
+    else
+    {
+        s.append(clientAddress()).append(":").append(std::to_string(clientPort()));
+    }
+    return s;
+}
+
+std::string Socket::getStatsString(std::chrono::steady_clock::time_point now) const noexcept
+{
+    const auto durTotal = std::chrono::duration_cast<std::chrono::milliseconds>(now - _creationTime);
+    const auto durLast = std::chrono::duration_cast<std::chrono::milliseconds>(now - _lastSeenTime);
 
-bool StreamSocket::socketpair(std::shared_ptr<StreamSocket> &parent,
-                              std::shared_ptr<StreamSocket> &child)
+    float kBpsIn, kBpsOut;
+    if (durTotal.count() > 0)
+    {
+        kBpsIn = (float)_bytesRcvd / (float)durTotal.count();
+        kBpsOut = (float)_bytesSent / (float)durTotal.count();
+    }
+    else
+    {
+        kBpsIn = (float)_bytesRcvd / 1000.0f;
+        kBpsOut = (float)_bytesSent / 1000.0f;
+    }
+    return Util::formatString("Socket[#%d, dur[total %" PRIi64 "ms, last %" PRIi64
```

Comment:
This seems overkill and in the wrong direction. Surely we don't need (vs)printf-style formatting. In this particular case, we can have a helper that creates this string with the proper formatting using `stringstream` or some such.

What's the motivation behind writing `vformatString()` at all?

Reply:
quoting myself ..
> I just like printf formatting as it is more performant and flexible (at least to me).
> IMHO I use it in a most reasonable save fashion, while being supported by the compiler to
> remove the risk of wrong format specifiers and stack values. The latter is still a weakness, which I am aware of.

TBH .. probably more a personal style. If the project rejects the `printf` formatting in general, I will drop it ofc.